STM: added an option to only compute dV STM

vvittaldev opened the pull request:
A Work in Progress diff that allows us to compute the STM with respect to only the velocity for most of our realistic cases where we can only use dV to target.

Still need to add test and linting

samotiwala commented:
Well I don't understand what the coverage/coveralls failure means...but the rest looks good to me!

vvittaldev commented:
I added the test for the velocity differentiation part in the stm_propagation_module_test.py file and that seems to have slightly increased the test coverage and removed that error 😄

vvittaldev commented:
Looks like adding a test in stm_propagation_module_test.py did the trick!
